### PR TITLE
DashboardScene: Integrate annotations

### PR DESCRIPTION
**Pull request by dprokop**
Integrates https://github.com/grafana/scenes/pull/325, https://github.com/grafana/scenes/pull/328, https://github.com/grafana/scenes/pull/329 to provide Annotations support in Dashboard Scene

TODO:
- [x] Add scene -> dashboard JSON transformation

### CODE REVIEW

File: public/app/features/dashboard-scene/utils/createPanelDataProvider.ts [hunk 1]
```diff
@@ -24,6 +24,9 @@ export function createPanelDataProvider(panel: PanelModel): SceneDataProvider |
     dataProvider = new SceneQueryRunner({
       queries: panel.targets,
       maxDataPoints: panel.maxDataPoints ?? undefined,
+      dataLayerFilter: {
+        panelId: panel.id,
+      },
```
Comment (torkelo) [inline public/app/features/dashboard-scene/utils/createPanelDataProvider.ts:29]: One potentially tricky bit of logic is that the panelId filtering should only happen when 
* the annotation query has a filter.ids array 
* is of type "dashboard" (should only apply to annotation queries for the build in -- Grafana -- data source)

https://github.com/grafana/grafana/blob/main/public/app/features/query/state/DashboardQueryRunner/utils.ts#L71

One solution for that would be to only return annotation items with panelId when the query has type "dashboard" , and skip them / clear them when type is "tags"

File: public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts [hunk 2]
```diff
@@ -50,6 +54,9 @@ export function transformSaveModelToScene(rsp: DashboardDTO): DashboardScene {
     autoMigrateOldPanels: true,
   });
 
+  // Setting for built-in annotations query to run
+  getDashboardSrv().setCurrent(oldModel);
```
[the remark this answers is not in the document]
Comment (dprokop) [reply public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts:58]: It's used in the Grafana datasource `query` method unfortunatelly: https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/grafana/datasource.ts#L91

File: public/app/features/dashboard-scene/utils/createPanelDataProvider.ts (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard-scene/utils/createPanelDataProvider.ts:29]: @torkelo - yes, this is handled by the transformation imlpemented in https://github.com/grafana/scenes/pull/329

Comment (torkelo) [reply public/app/features/dashboard-scene/utils/createPanelDataProvider.ts:29]: nice!

File: public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts:58]: I see, I think we can rid of all of those uses (timezone should be taken from request obj, dashboardUID likewise)

[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts:58]: but we will need this getDashboardSrv().setCurrent for many other things so it's good to have, but probably need to change to some compatability proxy at some point

**Review by torkelo — APPROVED** [review]
Looks good!

**Pull request merged**